FLASH_ERASE_C_EXPLICIT define is not found

Hey all,

In `flash_map_backend.h` we have the following snippet

```
#if defined(CONFIG_MCUBOOT)
static inline bool flash_area_erase_required(const struct flash_area *fa)
{
#if defined(CONFIG_FLASH_HAS_EXPLICIT_ERASE) && defined(CONFIG_FLASH_HAS_NO_EXPLICIT_ERASE)
    const struct flash_parameters *fp = flash_get_parameters(flash_area_get_device(fa));

    return flash_params_get_erase_cap(flash_get_parameters(flash_area_get_device(fa))) &
            FLASH_ERASE_C_EXPLICIT;
#elif defined(CONFIG_FLASH_HAS_EXPLICIT_ERASE)
    return true;
#else
    return false;
#endif
}
#endif
```

When I build the app with both CONFIG_FLASH_HAS_EXPLICIT_ERASE and CONFIG_FLASH_HAS_NO_EXPLICIT_ERASE enabled I see

```
/Users/vanpetrosyan/Projects/scs-workspace/bootloader/mcuboot/boot/zephyr/include/flash_map_backend/flash_map_backend.h:117:13: error: 'FLASH_ERASE_C_EXPLICIT' undeclared (first use in this function)
  117 |             FLASH_ERASE_C_EXPLICIT;
```

Seems like we are missing  `FLASH_ERASE_C_EXPLICIT` define, which is defined in  `<zephyr/drivers/flash.h>`. When I add the file to the includes everything works.

--- a/boot/zephyr/include/flash_map_backend/flash_map_backend.h
+++ b/boot/zephyr/include/flash_map_backend/flash_map_backend.h
@@ -9,6 +9,7 @@
 #define __FLASH_MAP_BACKEND_H__
 
 #include <zephyr/storage/flash_map.h> // the zephyr flash_map
+#include <zephyr/drivers/flash.h>
 
 #ifdef __cplusplus
 extern "C" {

--- a/boot/zephyr/include/flash_map_backend/flash_map_backend.h
+++ b/boot/zephyr/include/flash_map_backend/flash_map_backend.h
@@ -112,13 +112,15 @@ int flash_area_get_sector(const struct flash_area *fa, off_t off,
 static inline bool flash_area_erase_required(const struct flash_area *fa)
 {
 #if defined(CONFIG_FLASH_HAS_EXPLICIT_ERASE) && defined(CONFIG_FLASH_HAS_NO_EXPLICIT_ERASE)
-    const struct flash_parameters *fp = flash_get_parameters(flash_area_get_device(fa));
-
     return flash_params_get_erase_cap(flash_get_parameters(flash_area_get_device(fa))) &
             FLASH_ERASE_C_EXPLICIT;
 #elif defined(CONFIG_FLASH_HAS_EXPLICIT_ERASE)
+    (void)fa;
+
     return true;
 #else
+    (void)fa;
+
     return false;
 #endif
 }

